int16 id
float32 X
float32 Y
